### PR TITLE
Add article about debugging in XOD

### PR DESCRIPTION
**Pull request by brusherru**

### CODE REVIEW

File: docs/guide/debugging/README.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+Here is a quick summary of XOD debugging features:
+
+- Use debug nodes for the debugging, they are available in the `xod/debug` library
+- Debug your program in the simulation mode or right on the board whatever makes more sense
+
+In addition, a few other opportunities are available:
+
+- You can use the Serial tab in the Deployment pane to communicate with your hardware. Hit Deploy -> Connect Serial... and select the port. This is useful to talk to a UART hardware directly to configure a module or better understand its protocol.
+- XOD supports automated testing, called [tabular tests](/docs/guide/testing-patches). This allows you to debug individual parts of the program faster.
```
Comment (nkrkv) [inline docs/guide/debugging/README.md]:
```suggestion
- XOD supports automated testing, called [tabular tests](/docs/guide/testing-patches). This feature allows you to debug individual parts of the program faster.
```

File: docs/guide/debugging/README.md [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+Now you can see the temperature from your thermometer right on the screen of your computer and on the LCD. Also, you can change the text on the LCD directly from your computer.
+
+## Drill-down
+
+When your program becomes complex, it's an excellent practice to [encapsulate small pieces](/docs/guide/nodes-for-xod-in-xod/) of the logic into separate patches and compose them on the higher level of the program. Soon you will need to debug something inside these small pieces to make sure that the whole program works fine.
+
+The panel on the top of the debugger tab shows a series of breadcrumbs. It's a path to the currently open patch in the debugger tab.
+
+![Screenshot of debugger breadcrumbs](./breadcrumbs.png)
```
Comment (nkrkv) [inline docs/guide/debugging/README.md]: I think a drilled-down state will better illustrate the feature here. Can you shot the breadcrumbs when we’re deep inside? Ex: simulation > thermometer-tmp36 > analog-sensor

**Comment by brusherru** [comment]
@nkrkv,
I've renamed "Project and libraries" section into "Development" and moved "Debugging programs" into this section.

File: docs/guide/README.md [hunk 3]
```diff
@@ -54,10 +54,12 @@ IoT and network communication:
 - [Connecting to Internet with ESP8266-based MCUs](./esp8266-connect/)
 - [Fetching data from web API’s with HTTP GET requests](./http-get/)
 
-## Projects and libraries
+<a name="projects-and-libraries"><!-- Old anchor name --></a>
+## Development
 
-How to create, manage, and share your work.
+How to create, manage and share programs in XOD.
```
Comment (nkrkv) [inline docs/guide/README.md]:
```suggestion
How to create, manage, and share programs in XOD.
```

File: docs/README.md [hunk 4]
```diff
@@ -1,5 +1,6 @@
 ---
 title: XOD Documentation
+version: 1.0.0
```
Comment (nkrkv) [inline docs/README.md]:
```suggestion
version: 1.1.0
```

The `1.0.0` is implied by the site engine AFAIK

**Pull request merged**